### PR TITLE
Fix #305

### LINKED ISSUE #305: Regression (memory usage) for rc1?
To reproduce (CPython 3.9, Linux, x86):
```
$ ulimit -v $((512*1024))
$ pip install -U --upgrade-strategy eager diofant[gmpy]
$ python
>>> import diofant
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sk/src/diofant/diofant/__init__.py", line 8, in <module>
    from .core import (Add, Atom, AtomicExpr, Basic, Catalan, Derivative, Dict,
  File "/home/sk/src/diofant/diofant/core/__init__.py", line 8, in <module>
    from .expr import Expr, AtomicExpr
  File "/home/sk/src/diofant/diofant/core/expr.py", line 4, in <module>
    from mpmath.libmp import mpf_log, prec_to_dps
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/__init__.py", line 5, in <module>
    from .ctx_fp import FPContext
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/ctx_fp.py", line 1, in <module>
    from .ctx_base import StandardBaseContext
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/ctx_base.py", line 3, in <module>
    from .libmp.backend import xrange
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/libmp/__init__.py", line 17, in <module>
    from .libmpc import (mpc_one, mpc_zero, mpc_two, mpc_half,
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/libmp/libmpc.py", line 23, in <module>
    from .libelefun import (\
  File "/home/sk/venv/dev/lib/python3.9/site-packages/mpmath/libmp/libelefun.py", line 76, in <module>
    cache_prec_steps += [min(2**k,LOG_TAYLOR_PREC)+20] * 2**(k-1)
MemoryError
```
Without gmpy2 this import works (try e.g. `DIOFANT_GROUND_TYPES=python python`).

### PR DESCRIPTION
**Pull request by skirpichev**
Second and 3rd commits are optional.

### CODE REVIEW

**Comment by skirpichev** [comment]
FYI: mpc-devel in the image is too old to use it.

File: scripts/before_ci_build.sh [hunk 1]
```diff
@@ -1,23 +1,27 @@
+GMP_VERSION=6.2.1
+MPFR_VERSION=4.1.0
+MPC_VERSION=1.2.1
 if [ ! -f finish_before_ci_build ]; then
-  if [[ "$OSTYPE" == "linux-gnu" ]]; then
-    # yum install -y wget lzip
-    # wget https://gmplib.org/download/gmp/gmp-6.2.0.tar.lz
-    # tar -xvf gmp-6.2.0.tar.lz
+  if [ "$OSTYPE" == "linux-gnu" -o "$OSTYPE" == "darwin"* ]; then
```
Comment (isuruf) [inline scripts/before_ci_build.sh:5]: `OSTYPE` is not set in CI. Can you do `$(uname) == "Darwin"`?

Comment (skirpichev) [reply scripts/before_ci_build.sh:5]: This variable was from bash, which run the before_ci_build.sh.  Why we can't use one?

BTW, does this pr fix #310?

Comment (isuruf) [reply scripts/before_ci_build.sh:5]: > BTW, does this pr fix #310?

No, because this condition is not met.

Comment (skirpichev) [reply scripts/before_ci_build.sh]: Indeed, wildcards don't work for the POSIX test utility.  Bashism restored.

**Comment by isuruf** [comment]
Works now on a macOS 10.9

**Comment by skirpichev** [comment]
@casevh, #310 seems fixed.  @isuruf, thank you for tests.

**Comment by casevh** [comment]
@skirpichev @isuruf Thank you for resolving these issues.

**Pull request merged**